src/runtime/internal/dom.ts

```diff
@@ -112,6 +112,36 @@ function init_hydrate(target: NodeEx) {
 	}
 }
 
+export function append_styles(
+	target: Node,
+	styleSheetId: string,
+	styles: string
+) {
+	const appendStylesTo = get_root_for_styles(target);
+
+	if (!appendStylesTo?.querySelector(`#${styleSheetId}`)) {
```

Comment:
Can we get-element-by-id here please? If someone decides to put weird characters in their stylesheet id, I believe they would have to be escaped in order to query for it with `#foo`.

Reply:
`getElementById` only works on the `document` node. The function `get_root_for_styles` will return `document.head` where we only can use `querySelector`. 

@Conduitry you are right about needing to escape the selector. About A year ago I needed exact the same thing. To solve this problem I had to add 450byte of code. So I assume this is not something we want.

I will rewrite the `append_styles` function to make use of the `getElementById` function.

Reply:
A saw that `getElementById` also exists on `ShadowRoot`. I updated the code.